utils/build/test.mk

```diff
@@ -0,0 +1,94 @@
+
+# This is like build.mk but for test files
+T_CFLAGS  := $(CFLAGS)
+T_LFLAGS  := $(LFLAGS)
+T_INC_DIR := $(INC_DIR)
+T_TARGET  := testrunner
+T_MAIN    := main.test.cpp
+T_OBJ_DIR := $(OBJS_DIR)/$(TEST_DIR)
+
+# defines T_SRCS
+include $(TEST_DIR)/$(SRCS_DIR)/Test.files
+T_SRCS      += $(T_MAIN)
+T_OBJS      := $(patsubst %.cpp,$(T_OBJ_DIR)/%.o,$(T_SRCS))
+T_DEPFLAGS   = -MT $@ -MMD -MP -MF $(T_OBJ_DIR)/$*.d
+
+COVERAGE_FLAGS  := --coverage
+
+# libraries
+GTEST_DIR     := $(TEST_DIR)/$(LIBS_DIR)/googletest
+GTEST_TARGET  := $(GTEST_DIR)/build/lib/libgtest.a
+T_INC_DIR     += -I$(GTEST_DIR)/googletest/include
+T_LFLAGS      += -L$(dir $(GTEST_TARGET)) -lgtest
+GTEST_FILTERS +=
+
+CPPCHECK_DIR    := $(TEST_DIR)/$(LIBS_DIR)/cppcheck
+CPPCHECK_TARGET := $(CPPCHECK_DIR)/cppcheck
+CPPCHECK_EXEC   := $(shell command -v cppcheck || echo ./$(CPPCHECK_TARGET))
+CPPCHECK_FLAGS=--quiet --error-exitcode=1 --inline-suppr --suppressions-list=$(TEST_DIR)/lib/cppcheck-suppress
+
+# dependency checks
+## Is 1 when python is installed
+PYTHON=$(shell python2.7 -V  >/dev/null 2>&1 && echo "1" )
+
+PYTHONCHECK=$(shell [[ -z "$(PYTHON)" ]] && echo "0" || echo "1")
+
+# Manual override for LINTER
+# run "make VERBOSE=1" to see all commands
+ifndef NOLINT
```

Comment:
How come `NOLINT` get moved out to `test.mk`? Isn't overriding the linter applicable to everyone and not just testing?

Reply:
This should be removed, it was mistakenly left here after copy-paste from the original testing makefile.